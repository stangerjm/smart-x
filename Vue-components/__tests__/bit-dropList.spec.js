import {mount} from 'vue-test-utils'
import bitDropList from '../bit-dropList'
import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'

let dropItems = [
    {
      'itemTitle':'Home',
      'linkPath':'#'
    },
    {
      'itemTitle':'WSP',
      'linkPath':'#'
    },
    {
      'itemTitle':'About',
      'linkPath':'#'
    }
  ];

function mountDropList(props) {
  if (!props) {
    props = {
      propsData: {
        dropItems: dropItems
      }
    }
  }

  return mount(bitDropList, props);
}

describe('bit-dropList.vue', () => {
  it('has "bit-dropList" as a class value', () => {
    const wrapper = mountDropList();
    expect(wrapper.vm.$el.className).toBe('bit-dropList');
  });

  it('requires an array to be passed in to the dropItems property.', () => {
    const wrapper = mountDropList();

    let dropItemsProp = wrapper.vm.$options.props.dropItems;
    expect(dropItemsProp.type).toEqual(Array);
    expect(dropItemsProp.required).toBeTruthy();
  });

  it('renders all of the items passed in correctly.', () => {
    const wrapper = mountDropList();

    let dropListItems = wrapper.findAll('a');
    for(let dropItem of dropListItems.wrappers) {
      let itemText = dropItem.element.innerHTML;
      let itemHref = dropItem.element.getAttribute('href');

      let found = dropItems.find((item) => {
        return item.itemTitle === itemText
      });

      expect(found).not.toBeNull();
      expect(found.linkPath).toEqual(itemHref);
    }
  });

  it('has same HTML structure', () => {
    const renderer = createRenderer();
    const wrapper = shallow(bitDropList, {
      propsData: {
        dropItems: dropItems
      }
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err);
      }
      expect(str).toMatchSnapshot();
    });
  });
});