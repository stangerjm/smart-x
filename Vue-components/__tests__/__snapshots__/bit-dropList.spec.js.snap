// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bit-dropList.vue has same HTML structure 1`] = `
<ul class="bit-dropList">
  <li>
    <a href="#" class="wsp-link">Home</a>
  </li>
  <li>
    <a href="#" class="wsp-link">WSP</a>
  </li>
  <li>
    <a href="#" class="wsp-link">About</a>
  </li>
</ul>
`;
